### PR TITLE
Feature/user info

### PR DESCRIPTION
**Pull request by H-Bo214**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing - building tests
- [ ] Formatting - no new features
- [ ] Styling- no new features
### Detailed Description (Why is this change required? What problem does it solve? Any challenges that arose while implementing this feature? If so, how were the addressed?)

* Update user  back to currentUser
* setState for user when logged in to get name and photo
* Added conditional rendering of login-logout button in header
* Passed down props of currentUserName and logOutCurrentUser method to Header
* Passed currentUserName prop to Login and changed Login from a class component to functional component
* Fix logout from google error
* Update callback in uiConfig from signInSuccess to signInSuccessWithAuthResult as signInSuccess is now deprecated

### Files modified:
- [ ] index.html
- [x] App
- [ ] CardContainer
- [ ] PokeCard
- [ ] PokeForm
- [x] Login
- [x] FormContainer
- [ ] index.js
- [ ] Other files:

### CODE REVIEW

File: stretch-auth/src/App/App.js [hunk 1]
```diff
@@ -1,118 +1,122 @@
...
-  ], 
+    firebase.auth.EmailAuthProvider.PROVIDER_ID,
+  ],
   callbacks: {
-    signInSuccess: () => false
-  }
-}
+    signInSuccessWithAuthResult: () => false,
+  },
+};
```
Comment (jordy1611) [inline stretch-auth/src/App/App.js]: Feel like we just basically copy and paste this every time we want to do oauth

**Review by jordy1611 — COMMENTED** [review]
Really good work!

**Review by AurumValian — APPROVED** [review]
We definitely need to resolve these merge conflicts. That said, awesome setup of Firebase!

**Review by nware1066 — APPROVED** [review]
This is great work, I think this brings us up to MVP already!

**Pull request merged**